kamon-fluentd/src/main/scala/kamon/fluentd/Fluentd.scala

```diff
@@ -0,0 +1,144 @@
+import kamon.metric._
+import kamon.metric.instrument.{ Counter, Histogram }
+import kamon.util.ConfigTools.Syntax
+import kamon.util.MilliTimestamp
+import org.fluentd.logger.scala.FluentLogger
+import org.fluentd.logger.scala.sender.{ ScalaRawSocketSender, Sender }
+
+import scala.collection.JavaConverters._
+import scala.concurrent.duration.FiniteDuration
+
+object Fluentd extends ExtensionId[FluentdExtension] with ExtensionIdProvider {
+  override def lookup(): ExtensionId[_ <: Extension] = Fluentd
+
+  override def createExtension(system: ExtendedActorSystem): FluentdExtension = new FluentdExtension(system)
+}
+
+class FluentdExtension(system: ExtendedActorSystem) extends Kamon.Extension {
+  private val fluentdConfig = system.settings.config.getConfig("kamon.fluentd")
+  val host = fluentdConfig.getString("hostname")
+  val port = fluentdConfig.getInt("port")
+  val tag = fluentdConfig.getString("tag")
+  val flushInterval = fluentdConfig.getFiniteDuration("flush-interval")
+  val tickInterval = Kamon.metrics.settings.tickInterval
+  val subscriptions = fluentdConfig.getConfig("subscriptions")
+
+  val log = Logging(system, classOf[FluentdExtension])
+  log.info("Starting the Kamon(Fluentd) extension")
+
+  val subscriber = buildMetricsListener(tickInterval, flushInterval)
+  subscriptions.firstLevelKeys foreach { subscriptionCategory ⇒
+    subscriptions.getStringList(subscriptionCategory).asScala.foreach { pattern ⇒
+      Kamon.metrics.subscribe(subscriptionCategory, pattern, subscriber, permanently = true)
+    }
+  }
+
+  def buildMetricsListener(tickInterval: FiniteDuration, flushInterval: FiniteDuration): ActorRef = {
+    assert(flushInterval >= tickInterval, "Fluentd flush-interval needs to be equal or greater to the tick-interval")
+
+    val metricsSender = system.actorOf(Props(new FluentdMetricsSender(tag, host, port)), "kamon-fluentd")
+    if (flushInterval == tickInterval) {
+      metricsSender
+    } else {
+      system.actorOf(TickMetricSnapshotBuffer.props(flushInterval, metricsSender), "kamon-fluentd-buffer")
+    }
+  }
+}
+
+class FluentdMetricsSender(val tag: String, val host: String, val port: Int)
+    extends Actor with ActorLogging with FluentLoggerSenderProvider {
+
+  import kamon.fluentd.FluentdMetricsSender.RichHistogramSnapshot
+
+  val appName = context.system.settings.config.getString("kamon.fluentd.application-name")
+
+  lazy val fluentd = FluentLogger(tag, sender(host, port))
+
+  def receive = {
+    case tick: TickMetricSnapshot ⇒ sendMetricSnapshotToFluentd(tick)
+  }
+
+  def sendMetricSnapshotToFluentd(tick: TickMetricSnapshot): Unit = {
+    val time = tick.to
+    for {
+      (groupIdentity, groupSnapshot) ← tick.metrics
+      (metricIdentity, metricSnapshot) ← groupSnapshot.metrics
+    } {
+
+      val label = metricNameFor(groupIdentity, metricIdentity)
+
+      metricSnapshot match {
+        case hs: Histogram.Snapshot ⇒
+          if (hs.numberOfMeasurements > 0) {
```

Comment:
@dpsoft I finished working "configurable histogram stats report".  Configuration will be like bleow:
```
kamon.fluentd {
  # statistic values to be reported for histogram type metrics
  # (i.e. Histogram, MinMaxCounter, Gauge).
  histogram-stats {
    # stats values:
    # "count", "min", "max", "average", "percentiles" are supported.
    # you can use "*" for wildcards.
    subscription = [ "count", "min", "max", "average", "percentiles" ],

    # percentile points:
    # this will be used when you set "percentiles" in "subscription" above.
    # In this example, kamon-fluentd reports 50th 90th, 99th and 99.9th percentiles.
    percentiles = [50.0, 90.0, 99.0, 99.9]
  }
}
```
I will push squashed commit soon.  Please take a look at it.